src/primitives/box.jl

```diff
@@ -33,6 +33,23 @@ measure(b::Box) = prod(b.max - b.min)
 diagonal(b::Box) = norm(b.max - b.min)
 sides(b::Box) = b.max - b.min
 
+"""
+Return all corners of the box.
+"""
+@generated function vertices(b::Box{Dim,T}) where {Dim,T}
+  list = CartesianIndices(Tuple(repeat([1:2], Dim)))
```

Comment:
Maybe `CartesianIndices(ntuple(i->2, Dim))` to avoid heap allocations.

Reply:
That's cleaner too, nice.